src/Microsoft.Azure.WebJobs.Host.Storage/BlobStorageConcurrencyStatusRepository.cs

```diff
@@ -70,12 +70,12 @@ public async Task WriteAsync(HostConcurrencySnapshot snapshot, CancellationToken
 
             try
             {
-                CloudBlobContainer? container = await GetContainerAsync(cancellationToken);
+                BlobContainerClient? container = await GetContainerAsync(cancellationToken);
                 if (container != null)
                 {
-                    CloudBlockBlob blob = container.GetBlockBlobReference(blobPath);
+                    BlockBlobClient blob = container.GetBlockBlobClient(blobPath);
 
-                    using (StreamWriter writer = new StreamWriter(await blob.OpenWriteAsync(cancellationToken)))
+                    using (StreamWriter writer = new StreamWriter(await blob.OpenWriteAsync(overwrite: true, cancellationToken: cancellationToken)))
```

Comment:
Any reason for the differing implementation from https://github.com/Azure/azure-functions-host/blob/8b5973c2c07e4ae1bb2e46de4453a58c41e600ea/src/WebJobs.Script/Scale/BlobStorageConcurrencyStatusRepository.cs#L89?